pkg/drivers/kic/kic.go

```diff
@@ -0,0 +1,183 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kic
+
+import (
+	"fmt"
+
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/state"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/net"
+	pkgdrivers "k8s.io/minikube/pkg/drivers"
+	"k8s.io/minikube/pkg/minikube/command"
+	"k8s.io/minikube/pkg/minikube/constants"
+	"k8s.io/minikube/pkg/minikube/cruntime"
+)
+
+const driverName = constants.DriverKic
+
+// Driver is a driver designed to run kubeadm inside a container
+type Driver struct {
+	*drivers.BaseDriver
+	*pkgdrivers.CommonDriver
+	URL       string
+	runtime   cruntime.Manager
+	exec      command.Runner
+	OciClient string // docker or podman
+	KicImage  string
+}
+
+// Config is configuration for the kic driver
+type Config struct {
+	MachineName      string
+	StorePath        string
+	ContainerRuntime string
+	OciClient        string
+	KicImage         string // Image used for nodes
+}
+
+// NewDriver returns a fully configured None driver
+func NewDriver(c Config) *Driver {
+	runner := &command.OciRunner{} // MEDYA:TODO pass the node selector
+	// runtime, err := cruntime.New(cruntime.Config{Type: c.ContainerRuntime, Runner: runner})
+	// Libraries shouldn't panic, but there is no way for drivers to return error :(
+	// if err != nil {
+	// 	glog.Fatalf("unable to create container runtime: %v", err)
+	// }
+	d := &Driver{
+		BaseDriver: &drivers.BaseDriver{
+			MachineName: c.MachineName,
+			StorePath:   c.StorePath,
+		},
+		CommonDriver: &pkgdrivers.CommonDriver{},
+		exec:         runner,
+		KicImage:     c.KicImage,
+		OciClient:    c.OciClient,
+	}
+
+	fmt.Printf("MEDYA: Creating a New Driver %v", d)
+	return d
+}
+
+// PreCreateCheck checks for correct privileges and dependencies
+func (d *Driver) PreCreateCheck() error {
+	err := d.runtime.Available()
+	if err != nil {
+		return errors.Wrap(err, "pre-check for kic driver")
+	}
+	// TODO check if oci is in path
+	return nil
+}
+
+// Create a host using the driver's config
+func (d *Driver) Create() error {
+	// creation for the none driver is handled by commands.go
+	return nil
+}
+
+// DriverName returns the name of the driver
+func (d *Driver) DriverName() string {
+	return driverName
+}
+
+// GetIP returns an IP or hostname that this host is available at
+func (d *Driver) GetIP() (string, error) {
+
+	// MEDYA:TODO
+	// if alread exists use docker inspect --format '{{ .NetworkSettings.IPAddress }}' d.MachineName
+	ip, err := net.ChooseBindAddress(nil)
```

Comment:
Hmm, so this is returning the docker host address. I would have expected a docker address like 172.17.0.2

Or at least localhost, rather than the default bind address. This is similar to the current issues with `none`

Reply:
I actually dont know what would be best here. what are the current issues with none binding address ?

do you think it is better to bind on localhost ? or on the private ip that machine has ?

I was thinking we let the user pass the ip if they want to. so it doesnt have to be localhost. for mroe advanced options ! 

I am all ears for any recommednaiton on the bind address

Reply:
https://github.com/kubernetes/minikube/issues/4313